partcad-ide-standalone/tools/brand.py

```
@@ -111,8 +111,21 @@ def brand_shim(shim_path: pathlib.Path, old_name: str, new_name: str, allow_miss
+    as well. A binary names its executable in its own way, if at all, and a
+    same-length textual substitution inside one is not something to attempt: the
+    rename is the whole of the branding it gets. Treat it exactly like a script
+    that does not mention the old name, which is what `--allow-missing` is for.
     """
-    text = shim_path.read_text(encoding="utf-8")
+    try:
+        text = shim_path.read_text(encoding="utf-8")
+    except UnicodeDecodeError:
+        if allow_missing:
+            return False
+        raise SystemExit(f"error: {shim_path} is not a text launcher; there is nothing to rewrite in it")
```

Ruff (0.16.1)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)